pkgs/build-support/trivial-builders/default.nix

```
@@ -299,7 +299,7 @@ rec {
         # GHC (=> shellcheck) isn't supported on some platforms (such as risc-v)
         # but we still want to use writeShellApplication on those platforms
         let
-          shellcheckSupported = lib.meta.availableOn stdenv.buildPlatform shellcheck-minimal.compiler;
+          shellcheckSupported = (builtins.tryEval shellcheck-minimal.outPath).success;
```

I think we should keep the availableOn check since it is the intended way and what we're doing is kind of hacky because whether something fails to evaluate can also be influenced by `broken` etc. which is configurable.

Also all checks should be performed on the compiler.

```suggestion
          shellcheckSupported = lib.meta.availableOn stdenv.buildPlatform shellcheck-minimal.compiler && (builtins.tryEval shellcheck-minimal.compiler.outPath).success;
```